src/api/entities/Identity/AssetPermissions.ts

```diff
@@ -350,6 +353,32 @@ export class AssetPermissions extends Namespace<Identity> {
     return optionize(middlewareEventToEventIdentifier)(event);
   }
 
+  /**
+   * Retrieve the identifier data (block number, date and event index) of the event that was emitted when this Identity was enabled/added as
+   *   an Agent with permissions over a specific Asset
+   *
+   * @note uses the middlewareV2
+   * @note there is a possibility that the data is not ready by the time it is requested. In that case, `null` is returned
```

Comment:
I don't fully understand this comment, I get there is eventual consistency between chain and middleware, but wouldn't the case of the Asset simply not existing be hard to distinguish between it will eventually be made.